### PR TITLE
u4-8361 From ids to guids

### PR DESCRIPTION
**Pull request by zpqrtbnk**

### CODE REVIEW

File: src/Umbraco.Core/Persistence/Repositories/RedirectUrlRepository.cs [hunk 1]
```diff
@@ -176,5 +193,13 @@ public IEnumerable<IRedirectUrl> GetAllUrls(int rootContentId, long pageIndex, i
             var rules = result.Items.Select(Map);
             return rules;
         }
+
+        private static string HashUrl(string url)
+        {
+            var h = new MD5CryptoServiceProvider();
+            var i = Encoding.UTF8.GetBytes(url);
+            var o = h.ComputeHash(i);
+            return Encoding.UTF8.GetString(o);
```
Comment (nul800sebastiaan) [inline src/Umbraco.Core/Persistence/Repositories/RedirectUrlRepository.cs]: So easy to give them proper variable names.. :)

Comment (nul800sebastiaan) [inline src/Umbraco.Core/Persistence/Repositories/RedirectUrlRepository.cs]: CTRL+R+R in R# ;-)

**Comment by nul800sebastiaan** [comment]
Okay, looks good, works fine. Although it's funny, I'm not a fan of the column name "hurl", I'd just make it `hashUrl` so it's readable. 
Left some comments on unreadable variable names as well. My OCD is kicking in ;-)

Just one question: I don't really understand the purpose of the Hashed URL, care to explain?

File: src/Umbraco.Core/Persistence/Repositories/RedirectUrlRepository.cs (hunk 1 above)
Comment (zpqrtbnk) [inline src/Umbraco.Core/Persistence/Repositories/RedirectUrlRepository.cs]: Renaming ;-)

**Comment by zpqrtbnk** [comment]
Renaming hurl into urlHash. Purpose is explained in some notes in RedirectUrlDto - the idea is that we cannot use the full url in a SQL index because it's 2048 chars and an index is max 900 chars, so we hash the url into 16 chars and use that hash in an index. Turns out SQL loves it, and the over-cost of hashing is little compared to the benefit when running queries.

**Comment by nul800sebastiaan** [comment]
Looks great! Nice little hack around the migration problems :+1:

**Comment by nul800sebastiaan** [comment]
Oh and it all works now, with beta tables, with a `hurl` column. It straightens out in the end.

**Pull request merged**